src/core/BaseNode.js

```diff
@@ -245,5 +178,72 @@ export default Class(null, {
    * @method exit
    * @param {Tick} tick A tick instance.
    **/
-  exit: function(tick) {},
-});
+  exit(tick) {}
+};
+
+/**
+ * Node ID.
+ * @property {string} id
+ * @readonly
+ **/
+BaseNode.prototype.id = null;
```

Comment:
Unfortunately this code contains a very subtle bug. Because the prototype is shared amongst all instances, if you change the `id` for one of them, all of the other instances will change too. More on this: https://stackoverflow.com/questions/16911132/how-are-properties-shared-across-instances-using-javascript-prototypes

Consider moving this as an instance variable:

```js
class B {
  constructor() {
    this.id = null;
  }
}
```

Ironically, moving all the properties to be instance properties will make the code much terse.